### PR TITLE
Extracted CMX3600 adapter and related sample data and tests

### PR DESCRIPTION
**Pull request by apetrynet**
**Link the Issue(s) this Pull Request is related to.**
This PR is related to issue #1386

**Summarize your change.**
This PR removes the `cmx_3600` adapter and related sample files and tests. 
There might still be some bits and pieces left, like in the documentation.

### CODE REVIEW

**Comment by jminor** [comment]
This looks pretty good to me. If you want to be really clean about this change we _could_ separate the `imp` -> `importlib` change out and land that into `main` before the adapter switchero. Same for the tests that use `otio_json` instead of `cmx_3600`. That would make this PR just the removal of the EDL adapter.

File: tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
     "adapters": [
         {
             "OTIO_SCHEMA" : "Adapter.1",
-            "name" : "cmx_3600",
+            "name" : "otio_json",
```
Comment (apetrynet) [inline tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json:6]: @jminor, is this what you meant regarding "otio_json"?
I believe this is a part of testing the plugin system and not directly related to the cmx adapter. Therefore I just updated it to something that will remain after the cmx extarction.

**Comment by apetrynet** [comment]
What do we do about references to the cmx adapter in the documentation? Should we do that in this PR or do a proper doc overhaul when we're done with all the adapters?

File: tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json (hunk 1 above)
Comment (jminor) [reply tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json:6]: Yes, I was just suggesting that this portion of the change could be landed into main if that helps make the extract_adapters changes easier to compare. It is fine as-is though :)

**Comment by jminor** [comment]
Lets do the documentation change after we have the adapters split out, but before landing the extract_adapters branch into main.

File: tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json (hunk 1 above)
Comment (apetrynet) [reply tests/baselines/plugin_module/otio_override_adapter/plugin_manifest.json]: On it! Thanks!

**Comment by apetrynet** [comment]
Looks like we need to update some of the docs after all to pass the tests.

**Review by jminor — APPROVED** [review]

**Pull request merged**